mdn/developer-portal
# 1539: Show applied filters and search params in mobile viewport

This changeset improves the mobile-viewport UX of filtering and searching the Posts, Events and People pages by showing a summary of any applied filters or search terms above the "Filters" panel, which is collapsed by default in narrow viewports

<img width="436" alt="Screenshot 2020-06-18 at 22 02 48" src="https://user-images.githubusercontent.com/101457/85277006-9199b180-b47a-11ea-9c6b-e6d77bdc33ad.png">

It also adds in an extra header above the filtering form for the Events page, because the up-page content on Events means that - on mobile - the filter form is a long way away from the page header, so adding in a custom header for Events gives the user a reminder of the context of what's going on. (At least, I think so and hope you agree!) - this issue was reported in #1594

<img width="449" alt="Screenshot 2020-06-22 at 11 15 29" src="https://user-images.githubusercontent.com/101457/85277144-d6254d00-b47a-11ea-9357-86eff4f35df9.png">

Finally, it also fixes an oversight in configuring Events for search in #1530 -- the `country` field also needed specifying as a wagtailsearch `FilterField` to prevent a 500.

(Related issues #1530 and #1594)

## How to test

- The code has been deployed to staging. 

- Please try using the filters and search parameters on all of the the following URLs in responsive mode (https://developer.mozilla.org/en-US/docs/Tools/Responsive_Design_Mode) where you will see the summary and then in desktop viewport, where you will not see it (nor the extra "Events" header above the filtering)

    - https://developer-portal.stage.mdn.mozit.cloud/events/
    - https://developer-portal.stage.mdn.mozit.cloud/communities/people/
    - https://developer-portal.stage.mdn.mozit.cloud/posts/

- Try adding content that needs escaping (eg `<script>alert("boo");<script>` or an accented name or term) into the search box and confirm that it is shown correctly in the summary of search terms and filters. View source if you can to confirm that it is rendered out as escaped characters (eg `&lt;script&gt;` etc)

## Screenshots

<img width="466" alt="Screenshot 2020-06-22 at 11 10 41" src="https://user-images.githubusercontent.com/101457/85278717-361cf300-b47d-11ea-9675-7326a05c4744.png">
<img width="478" alt="Screenshot 2020-06-22 at 11 14 31" src="https://user-images.githubusercontent.com/101457/85278722-36b58980-b47d-11ea-9b02-fc827817a5f6.png">
<img width="449" alt="Screenshot 2020-06-22 at 11 15 29" src="https://user-images.githubusercontent.com/101457/85278726-374e2000-b47d-11ea-9fd7-709712b45917.png">
